### ISSUE TITLE
test_keys flaky with randomized hashes

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5748]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)| @pitrou reported|
|-|-|
|Trac ID|trac#5748|
|Type|defect|
|Created|2012-07-08 21:08:37Z|

One test depends on hash ordering and fails sporadically with ``python -R``:

```
[FAIL]
Traceback (most recent call last):
  File "/home/antoine/twisted/twisted/conch/test/test_keys.py", line 498, in test_repr
    \t04>""")
  File "/home/antoine/twisted/twisted/trial/unittest.py", line 270, in assertEqual
    % (msg, pformat(first), pformat(second)))
twisted.trial.unittest.FailTest: not equal:
a = '<RSA Private Key (0 bits)\nattr p:\n\t04\nattr q:\n\t05\nattr u:\n\t04\nattr n:\n\t01\nattr d:\n\t03\nattr e:\n\t02>'
b = '<RSA Private Key (0 bits)\nattr e:\n\t02\nattr d:\n\t03\nattr n:\n\t01\nattr q:\n\t05\nattr p:\n\t04\nattr u:\n\t04>'
```

Attachments:

* [randomhash_keyrepr.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/85e/85ed038ac2e3e244786cc34667f7f3b31234aa08/c80edf92fb6e7c4d6ba0a63e38cf368e52ffe56a.patch) (937 bytes) - added by antoine on 2012-07-08 21:10:06Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5748 5748
type__defect defect
reporter__antoine antoine
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1341781717000000 1341781717000000
changetime__1341785649000000 1341785649000000
version__None None
owner__exarkun exarkun
cc__antoine
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)<a name="note_1"></a>|@pitrou commented|
|-|-|

Patch attached.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @exarkun|
|-|-|

Thanks.  This looks fine.  I'll apply.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [34702]) Adjust the repr string of conch Key objects to present deterministic results, regardless of dict item ordering.

Author: antoine
Reviewer: exarkun
Fixes: [#5748](https://github.com/twisted/twisted/issues/5748)
